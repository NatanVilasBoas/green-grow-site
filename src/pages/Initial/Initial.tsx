import { useEffect } from "react";
import CtaButton from "../../components/CtaButton/CtaButton";
import styled from "styled-components";
import CardBeneficio from "../../components/CardBeneficio/CardBeneficio";


const Container = styled.div`
    margin: 3vh 2vw;
`

const Beneficios = styled.section`
    display: flex;
    margin: 2vh 2vw;
    justify-content: space-between;
`

const Initial = () => {
    useEffect( () => {
            document.title = 'Inicial';
    })
    return (
        <Container>
            <h1>Bem-vindo à GreenGrow - Transformando a Agricultura Urbana</h1>
            <p>Seja bem-vindo à GreenGrow, sua fonte líder em soluções de agricultura urbana sustentável. Oferecemos uma variedade de produtos inovadores para ajudar você a cultivar alimentos frescos e saudáveis em seu próprio espaço urbano, de maneira fácil, sustentável e acessível.</p>

            <Beneficios role="group">
                <CardBeneficio 
                    title="Produtos de Qualidade" 
                    text="Nossos sistemas de cultivo vertical, compostagem doméstica e muito mais são projetados com os mais altos padrões de qualidade, garantindo resultados excepcionais." />
                <CardBeneficio 
                    title="Sustentabilidade" 
                    text="Estamos comprometidos com práticas agrícolas sustentáveis que beneficiam o meio ambiente e promovem comunidades mais saudáveis e resilientes." />
                <CardBeneficio 
                    title="Facilidade de Uso" 
                    text="Nossos produtos são projetados pensando na facilidade de uso e são perfeitos para iniciantes e entusiastas experientes da jardinagem." />
            </Beneficios>
           
            <CtaButton>Saiba mais</CtaButton>
            <CtaButton>Explore nossos produtos</CtaButton>
        </Container>
    )
}

export default Initial;