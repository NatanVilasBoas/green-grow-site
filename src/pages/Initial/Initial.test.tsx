import { fireEvent, render, screen } from "@testing-library/react"
import Initial from "./Initial";

const mockNavegacao = jest.fn();

jest.mock('react-router-dom', () => {
    return{
        ...jest.requireActual('react-router-dom'),
        useNavigate: () => mockNavegacao
    }
})

describe('A página inicial', () => {
    test('deve ser renderizada', () => {
        const {container} = render(<Initial/>)

        expect(container).toMatchSnapshot();
    })

    test('deve conter os botões CTA\'\s', () => {
        render(<Initial />)

        const botoes = screen.getAllByRole('button');
        botoes.forEach(botao => {
            expect(botao).toBeInTheDocument();
        })

    })

    test('deve conter os benefícios', () => {
        render(<Initial/>)

        const section = screen.getByRole('group-beneficios');

        expect(section).toBeInTheDocument();
    })

    test('deve conter os relatos de clientes', () => {
        render(<Initial/>)

        const relatoClientes = screen.getByRole('group-clientes');

        expect(relatoClientes).toBeInTheDocument();
    })

    test('os CTAs devem redirecionar a outras páginas', () => {
        render(<Initial/>)

        const buttons = screen.getAllByRole('button');

        buttons.forEach((button) => {
            fireEvent.click(button)
        })

        expect(mockNavegacao).toHaveBeenCalledTimes(buttons.length);
    })
})