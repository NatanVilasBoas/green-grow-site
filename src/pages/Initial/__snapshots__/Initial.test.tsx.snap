// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A página inicial deve ser renderizada 1`] = `
<div>
  <div
    class="sc-hLQSwg bgaAVe"
  >
    <h1>
      Bem-vindo à GreenGrow - Transformando a Agricultura Urbana
    </h1>
    <p>
      Seja bem-vindo à GreenGrow, sua fonte líder em soluções de agricultura urbana sustentável. Oferecemos uma variedade de produtos inovadores para ajudar você a cultivar alimentos frescos e saudáveis em seu próprio espaço urbano, de maneira fácil, sustentável e acessível.
    </p>
    <h3
      style="font-family: Montserrat;"
    >
      Por que escolher a GreenGrow?
    </h3>
    <section
      class="sc-eDLKkx iWMWPc"
      role="group"
    >
      <div
        class="sc-guDLey lphenX"
      >
        <h3
          class="sc-dmyCSP jxlpum"
        >
          Produtos de Qualidade
        </h3>
        <p>
          Nossos sistemas de cultivo vertical, compostagem doméstica e muito mais são projetados com os mais altos padrões de qualidade, garantindo resultados excepcionais.
        </p>
      </div>
      <div
        class="sc-guDLey lphenX"
      >
        <h3
          class="sc-dmyCSP jxlpum"
        >
          Sustentabilidade
        </h3>
        <p>
          Estamos comprometidos com práticas agrícolas sustentáveis que beneficiam o meio ambiente e promovem comunidades mais saudáveis e resilientes.
        </p>
      </div>
      <div
        class="sc-guDLey lphenX"
      >
        <h3
          class="sc-dmyCSP jxlpum"
        >
          Facilidade de Uso
        </h3>
        <p>
          Nossos produtos são projetados pensando na facilidade de uso e são perfeitos para iniciantes e entusiastas experientes da jardinagem.
        </p>
      </div>
    </section>
    <h3
      style="font-family: Montserrat;"
    >
      O que é a agricultura urbana sustentável?
    </h3>
    <p>
      É uma prática que envolve o cultivo de alimentos, plantas e outros produtos agrícolas em ambientes urbanos de maneira ecologicamente equilibrada, socialmente justa e economicamente viável. Ela busca promover a produção local de alimentos, reduzir a dependência de grandes cadeias de suprimentos e aumentar a resiliência das comunidades urbanas.
    </p>
    <h3
      style="font-family: Montserrat;"
    >
      O que nossos clientes dizem:
    </h3>
    <button
      class="sc-beySPh gSdwCo"
    >
      Saiba mais
    </button>
    <button
      class="sc-beySPh gSdwCo"
    >
      Explore nossos produtos
    </button>
  </div>
</div>
`;
